engine.py

```diff
@@ -198,7 +258,7 @@ def post_app_init_nuke(self, menu_name="Shotgun"):
         """
         The Nuke-specific portion of the engine's post-init process.
         """
-        if self.has_ui:
+        if self.has_ui and not self.studio_enabled:
```

Comment:
Looks like most of the code from post_app_init_{nuke,hiero} could be refactored as
_post_app_init(menu_name, tk_Nuke.NukeStudioMenuGenerator)
and
_post_app_init(menu_name, tk_Nuke.HieroMenuGenerator)